test/utils/cookie.test.ts

```diff
@@ -114,3 +114,33 @@ describe('createCookieObjectFromHeaderValue and createCookieStringFromObject tog
     expect(createCookieStringFromObject(...createCookieObjectFromHeaderValue(cookieString))).toBe(cookieString)
   })
 })
+
+describe('filterCookies', () => {
+  it('works when there is no cookie', () => {
+    const headers = new Headers()
+    headers.set('x', 'y')
+    const resultHeaders = filterCookies(headers, (key) => key === 'a')
+    expect(resultHeaders.get('cookie')).toBe(null)
+    expect(resultHeaders.get('x')).toBe('y')
+  })
+  it('removes other keys', () => {
+    const headers = new Headers()
+    headers.set('cookie', 'a=1; b=2')
+    const resultHeaders = filterCookies(headers, (key) => key === 'a')
+    expect(resultHeaders.get('cookie')).toBe('a=1')
+  })
+  it('removes other keys when no match', () => {
```

Comment:
Can you add one more test for case there were additional headers?

Reply:
Done in the next commit